### PR TITLE
[SNOW-1694712] Remove _emit_ast and _ast from docs.

### PR DESCRIPTION
**Pull request by sfc-gh-lspiegelberg**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR. 
   Fixes[SNOW-1694712](https://snowflakecomputing.atlassian.net/browse/SNOW-1694712)

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [x] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://github.com/snowflakedb/snowpark-python/blob/main/CONTRIBUTING.md#thread-safe-development)

4. Please describe how your code solves the related issue.

   AST introduces in function signatures (private) parameters `_emit_ast` and `_ast`. This PR modifies the sphinx-build to not emit these parameters into the docs. To check that the generation is successful, run e.e. from the `docs` folder
```
grep -rl --include "*.html" "_emit_ast" build/html
grep -rl --include "*.html" "_ast" build/html
```

### CODE REVIEW

**Comment by sfc-gh-aalam** [comment]
Screenshot 2024-12-03 at 10 03 12 AM

can you look at these formatting issues where we see double closed parenthesis

**Comment by sfc-gh-aalam** [comment]
Screenshot 2024-12-03 at 10 04 07 AM

there are similar issues I noticed in column and dataframe

File: docs/source/conf.py [hunk 1]
```diff
@@ -287,6 +283,24 @@ def process_modin_accessors(args):
...
         return items
 
+def process_signature(app, what, name, obj, options, signature, return_annotation):
+
+    # Names to remove from signature (AST related):
+    names_to_remove = ['_emit_ast', '_ast']
+
+    def remove_from_signature(signature, name_to_remove):
+        parts = [p for p in signature.split(',') if name_to_remove not in p]
+        signature = ','.join(parts)
```
Comment (sfc-gh-aling) [inline docs/source/conf.py:293]: I think the dup `))` is caused by single parameter signature 

for such case `def some_func(_emit_ast:bool=True)`, `signature` will be empty string on this line, and then on line 296, a single `)` is added

Comment (sfc-gh-lspiegelberg) [reply docs/source/conf.py:293]: good callout, let me take a look at this

Comment (sfc-gh-lspiegelberg) [reply docs/source/conf.py]: fixed by changing logic.

File: docs/source/conf.py [hunk 2]
```diff
@@ -287,6 +283,30 @@ def process_modin_accessors(args):
...
+            # temporarily remove parentheses, add after removing name_to_remove parts.
+            signature = signature[1:-1]
+            parts = [p for p in signature.split(',') if name_to_remove not in p]
+            signature = ','.join(parts)
+
+            return '(' + signature + ')'
+        else:
+            return signature
+
+    if signature and any(name_to_remove in signature for name_to_remove in ['_emit_ast', '_ast']):
```
Comment (sfc-gh-aalam) [inline docs/source/conf.py:305]: are we doing the same check twice to see if `_emit_ast`, `_ast` is in signature. I don't think this is necessary. We can remove the check here and make things simpler and easier to read.

File: docs/source/conf.py [hunk 3]
```diff
@@ -287,6 +283,30 @@ def process_modin_accessors(args):
...
+        if name_to_remove not in signature:
+            return signature
+
+        if signature.startswith('(') and signature.endswith(')'):
+            # temporarily remove parentheses, add after removing name_to_remove parts.
+            signature = signature[1:-1]
+            parts = [p for p in signature.split(',') if name_to_remove not in p]
+            signature = ','.join(parts)
+
+            return '(' + signature + ')'
```
Comment (sfc-gh-aalam) [inline docs/source/conf.py:301]: using f-strings is a better

**Review by sfc-gh-aling — COMMENTED** [review]
pre-approving, assuming you will address Afroz's comment

**Review by sfc-gh-aling — APPROVED** [review]

**Review by sfc-gh-aalam — APPROVED** [review]

**Pull request merged**